internal/provider/resource_cloudflare_tunnel_route.go

```diff
@@ -105,28 +124,40 @@ func resourceCloudflareTunnelRouteUpdate(ctx context.Context, d *schema.Resource
 
 func resourceCloudflareTunnelRouteDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*cloudflare.API)
+	network := d.Get("network").(string)
 
-	err := client.DeleteTunnelRoute(ctx, cloudflare.TunnelRoutesDeleteParams{
-		AccountID: d.Get("account_id").(string),
-		Network:   d.Get("network").(string),
-	})
+	resource := cloudflare.TunnelRoutesDeleteParams{
+		AccountID:        d.Get("account_id").(string),
+		Network:          network,
+		VirtualNetworkID: d.Get("virtual_network_id").(string),
+	}
+
+	err := client.DeleteTunnelRoute(ctx, resource)
 	if err != nil {
-		return diag.FromErr(fmt.Errorf("error deleting Tunnel Route for Network %q: %w", d.Get("network").(string), err))
+		return diag.FromErr(fmt.Errorf("error deleting Tunnel Route for Network %q: %w", network, err))
 	}
 
 	return nil
 }
 
 func resourceCloudflareTunnelRouteImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```

Comment:
this seems unnecessarily complex; let's just force it to include the `virtual_network_id` for explicitness.

Reply:
As a `cloudflared` user I know that vnet is totally optional param there.
Making it required during import will create additional headache for developers that are always working inside of the "default" vnet since it's not easy at all to get GUID of the default vnet (you cannot do it through UI. Only by using API or `cloudflared`).

But anyway for me it does not make any difference. So let's make it required.